core/dbt/cli/params.py

```diff
@@ -39,6 +39,16 @@
     default=True,
 )
 
+columns = click.option(
```

Comment:
When using Snowflake, it seems the columns need to be in capitals, which is also how the columns are output anyway to the screen, but would be good to note that columns might need to be capitalised, atleast until it's fixed in adapters (if needed(?).

When using lower case:
```sh
16:00:58  Encountered an error:
tuple.index(x): x not in tuple
16:00:58  Traceback (most recent call last):
  File "dbt-core/core/dbt/cli/requires.py", line 91, in wrapper
    result, success = func(*args, **kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^
  File "dbt-core/core/dbt/cli/requires.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "dbt-core/core/dbt/cli/requires.py", line 169, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "dbt-core/core/dbt/cli/requires.py", line 198, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "dbt-core/core/dbt/cli/requires.py", line 245, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "dbt-core/core/dbt/cli/requires.py", line 278, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "dbt-core/core/dbt/cli/main.py", line 420, in show
    results = task.run()
              ^^^^^^^^^^
  File "dbt-core/core/dbt/task/runnable.py", line 493, in run
    self.task_end_messages(result.results)
  File "dbt-core/core/dbt/task/show.py", line 88, in task_end_messages
    table = table.select(self.args.columns)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venvs/dbtsnowflake/.venv/lib/python3.11/site-packages/agate/table/select.py", line 21, in select
    indexes = tuple(self._column_names.index(k) for k in key)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venvs/dbtsnowflake/.venv/lib/python3.11/site-packages/agate/table/select.py", line 21, in <genexpr>
    indexes = tuple(self._column_names.index(k) for k in key)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: tuple.index(x): x not in tuple
```

Reply:
🤔 , just had a thought.

Just tried this, it works, but not sure if the variable should be moved out or it's fine to define here.

```python
                # convert columns to uppercase to match agate's behavior
                table = table.select([col.upper() for col in self.args.columns])
```

Reply:
Uppercasing all columns would break other databases. We could accept case-insensitve columns and map them to exact column names using `table.colum_names`. But that wouldn't work for people using case-sensitive column names.

To avoid the error we can filter out unknown columns (which includes incorrect casing) and add a note that exact column names are required:

```python
valid_columns = [col for col in self.args.columns if col in table.column_names]
table = table.select(valid_columns)
```